### PR TITLE
API-SDK version change and Fields addition in the schema. (#146)

### PR DESCRIPTION
**Pull request by KrisPersonal**
* TDL-17513: Upgrade API and SDK version (#135)

* added missing test cases

* updated sdk version and added new fields

* removed abandoned_checkouts from all test cases except discovery

* updated all fields test to remove value_type from metafields stream

* updated api and sdk version

* added comment for abandoned checkouts in test cases

* added value_type field in metafield containing type value

* resolved pylint error

* TDL-18045: Fields Addition in the Schema (#140)

* added fields in the schema

* updated schema of order_refunds

* updated arguments field schema

* removed unnecessary space in the schema

# Description of change
(write a short description here or paste a link to JIRA)

# QA steps
 - [ ] automated tests passing
 - [ ] manual qa steps passing (list below)
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_shopify/__init__.py [hunk 1]
```diff
@@ -167,7 +167,8 @@ def sync():
             Context.state['bookmarks'] = {}
         Context.state['bookmarks']['currently_sync_stream'] = stream_id
 
-        with Transformer() as transformer:
+        # some fields have epoch-time as date, hence transform into UTC date
+        with Transformer(singer.UNIX_SECONDS_INTEGER_DATETIME_PARSING) as transformer:
```
Comment (bryantgray) [inline tap_shopify/__init__.py:171]: Does this change the structure of existing datetime fields? Were these epoch-time fields previously replicated as epoch-time?

File: tap_shopify/schemas/definitions.json [hunk 2]
```diff
@@ -806,6 +869,189 @@
...
+        ]
+      },
+      "variant_price": {
+        "type": [
+          "null",
+          "string"
+        ],
+        "format": "singer.decimal"
+      },
+      "duties": {},
```
Comment (bryantgray) [inline tap_shopify/schemas/definitions.json:898]: Does this have a type?

File: tap_shopify/schemas/definitions.json [hunk 3]
```diff
@@ -806,6 +869,189 @@
...
+              "null",
+              "string"
+            ]
+          },
+          "pricing_policies": {
+            "type": [
+              "null",
+              "array"
+            ],
+            "items": {}
```
Comment (bryantgray) [inline tap_shopify/schemas/definitions.json:978]: Is there any other data that can be added to the items object? This goes for all the `"items": {}`.

File: tests/test_automatic_fields.py [hunk 4]
```diff
@@ -72,7 +79,15 @@ def automatic_test(self, conn_id, testable_streams):
                 # verify that only the automatic fields are sent to the target
                 self.assertEqual(
                     actual_fields_by_stream.get(stream, set()),
-                    self.expected_primary_keys().get(stream, set()) |
+                    expected_primary_keys |
```
Comment (bryantgray) [inline tests/test_automatic_fields.py:82]: Why was this changed to a variable, but not `self.expected_replication_keys().get(stream, set())`?

File: tap_shopify/schemas/definitions.json (hunk 2 above)
Comment (harshpatel4crest) [reply tap_shopify/schemas/definitions.json:898]: In the Shopify API [doc](https://shopify.dev/api/admin-rest/2022-01/resources/refund#resource-object) the duties is coming as an object, but from the API response, it is replicated as a list. Hence, used an empty schema here, to accept any value.

File: tap_shopify/__init__.py (hunk 1 above)
Comment (harshpatel4crest) [reply tap_shopify/__init__.py:171]: > Does this change the structure of existing datetime fields?

No, there is no change in the existing date fields.

> Were these epoch-time fields previously replicated as epoch-time?

No, we have added these fields. There is field named `created` in `order_refunds` whose response in coming in form of epoch time from the API.

File: tests/test_automatic_fields.py (hunk 4 above)
Comment (harshpatel4crest) [reply tests/test_automatic_fields.py:82]: We have created a variable named `expected_primary_keys` which contains the same value. Hence, updated the code to use the same variable here.

File: tap_shopify/schemas/definitions.json (hunk 3 above)
Comment (harshpatel4crest) [reply tap_shopify/schemas/definitions.json:978]: This will accept every data. We added this because we cannot find any doc in the Shopify API for the data-type of this field.

**Review by bryantgray — APPROVED** [review]
